Explorer/Assets/DCL/AvatarRendering/Loading/Systems/Abstract/LoadElementsByIntentionSystem.cs

```diff
@@ -0,0 +1,104 @@
+using Arch.Core;
+using CommunicationData.URLHelpers;
+using Cysharp.Threading.Tasks;
+using DCL.AvatarRendering.Loading.Components;
+using DCL.AvatarRendering.Loading.DTO;
+using DCL.AvatarRendering.Wearables.Components;
+using DCL.Optimization.PerformanceBudgeting;
+using DCL.WebRequests;
+using ECS;
+using ECS.Prioritization.Components;
+using ECS.StreamableLoading.Cache;
+using ECS.StreamableLoading.Common.Components;
+using ECS.StreamableLoading.Common.Systems;
+using System;
+using System.Threading;
+using Utility.Multithreading;
+
+namespace DCL.AvatarRendering.Loading.Systems.Abstract
+{
+    public abstract class LoadElementsByIntentionSystem<TAsset, TIntention, TAvatarElement, TAvatarElementDTO> :
+        LoadSystemBase<TAsset, TIntention>
+        where TIntention: struct, ICountedLoadingIntention<TAvatarElement>
+        where TAvatarElementDTO: AvatarAttachmentDTO where TAvatarElement : IAvatarAttachment<TAvatarElementDTO>
+    {
+        private readonly IAvatarElementCache<TAvatarElement, TAvatarElementDTO> avatarElementCache;
+        private readonly IWebRequestController webRequestController;
+        private readonly IRealmData realmData;
+
+        protected LoadElementsByIntentionSystem(
+            World world,
+            IStreamableCache<TAsset, TIntention> cache,
+            IAvatarElementCache<TAvatarElement, TAvatarElementDTO> avatarElementCache,
+            IWebRequestController webRequestController,
+            IRealmData realmData
+        ) : base(world, cache)
+        {
+            this.avatarElementCache = avatarElementCache;
+            this.webRequestController = webRequestController;
+            this.realmData = realmData;
+        }
+
+        protected sealed override async UniTask<StreamableLoadingResult<TAsset>> FlowInternalAsync(TIntention intention,
+            IAcquiredBudget acquiredBudget, IPartitionComponent partition, CancellationToken ct)
+        {
+            await realmData.WaitConfiguredAsync();
+
+            var lambdaResponse =
+                await ParsedResponseAsync(
+                    webRequestController.GetAsync(
+                        new CommonArguments(
+                            BuildUrlFromIntention(in intention),
+                            attemptsCount: intention.CommonArguments.Attempts
+                        ),
+                        ct,
+                        GetReportCategory()
+                    )
+                );
+
+            await using (await ExecuteOnThreadPoolScope.NewScopeWithReturnOnMainThreadAsync())
+                Load(ref intention, lambdaResponse);
+
+            return new StreamableLoadingResult<TAsset>(AssetFromPreparedIntention(in intention));
+        }
+
+        private void Load<TResponseElement>(ref TIntention intention, ILambdaResponse<TResponseElement> lambdaResponse) where TResponseElement: ILambdaResponseElement<TAvatarElementDTO>
+        {
+            intention.SetTotal(lambdaResponse.TotalAmount);
```

Comment:
What is the reason to set `TotalAmount` explicitly rather than using `List.Count`?

Reply:
I don't know the particular reason, I saved the previous logic here

Reply:
I'll dig deeper and try to get rid off of it